src/poprox_concepts/domain/click_history_with_timestamp

```diff
@@ -0,0 +1,8 @@
+from datetime import datetime
+from uuid import UUID
+
+from pydantic import BaseModel
+
+
+class ClickHistoryWithTimestamp(BaseModel):
+    article_id_with_time_tuples: list[(UUID, datetime)]
```

Comment:
A few questions/comments here:

- I think we prefer `List[...]` instead of `list[...]` (which you need to import from `typing` I think) 
- If we're using pydantic I think we would want all article types to be serializable to json and I don't know if `(UUID, datetime)` would be serializable or not. Did you test this?

Reply:
1. Sure, I can hold on pushing and try test this branch in the poprox-storage repo (I assume I could just uninstall poprox-concepts, and re-install it with this branch?). 

2. Thanks for catching the type! I referred to the lower-case `list()` we use in `click_history.py`, but agree typing `List` is better in that case. I'll change it.

3. Hmm yeah it's tricky if we can serialize tuple in python (I thought it should be, but after googling some looks like others transform tuple into dict first before serializing it to json). I can use `dict` type for safety!

Reply:
An alternate approach would be to go to create a model for capturing a single click, then have a list of those click models.

Reply:
I like the idea of having a `Click` model that includes the timestamp—it's usually better to create custom types with clear names than to shoehorn your data into Python primitives.

(The anti-pattern of using language primitives instead of classes/objects is sometimes called "primitive obsession," and there are a [bunch of strategies to avoid it](https://refactoring.guru/smells/primitive-obsession) depending on where it pops up.)

Reply:
Thank you both for the input, having a simple `Click` class also sounds good to me! I'll do that.